lanes/mev/abci.go

```diff
@@ -30,209 +30,228 @@ func (l *MEVLane) PrepareLaneHandler() base.PrepareLaneHandler {
+				if err != nil {
 					l.Logger().Info(
-						"failed to verify auction bid tx",
-						"tx_hash", hash,
+						"failed to get hash of bundled tx",
+						"tx_hash", txInfo.Hash,
 						"err", err,
 					)
 
-					txsToRemove = append(txsToRemove, tmpBidTx)
+					txsToRemove = append(txsToRemove, bidTx)
 					continue selectBidTxLoop
 				}
 
-				// store the bytes of each ref tx as sdk.Tx bytes in order to build a valid proposal
-				bundledTxBz := make([][]byte, len(bidInfo.Transactions))
-				for index, rawRefTx := range bidInfo.Transactions {
-					sdkTx, err := l.WrapBundleTransaction(rawRefTx)
-					if err != nil {
-						l.Logger().Info(
-							"failed to wrap bundled tx",
-							"tx_hash", hash,
-							"err", err,
-						)
-
-						txsToRemove = append(txsToRemove, tmpBidTx)
-						continue selectBidTxLoop
-					}
-
-					sdkTxBz, _, err := utils.GetTxHashStr(l.TxEncoder(), sdkTx)
-					if err != nil {
-						l.Logger().Info(
-							"failed to get hash of bundled tx",
-							"tx_hash", hash,
-							"err", err,
-						)
-
-						txsToRemove = append(txsToRemove, tmpBidTx)
-						continue selectBidTxLoop
-					}
-
-					// if the transaction is already in the (partial) block proposal, we skip it.
-					if proposal.Contains(sdkTxBz) {
-						l.Logger().Info(
-							"failed to select auction bid tx for lane; tx is already in proposal",
-							"tx_hash", hash,
-						)
-
-						continue selectBidTxLoop
-					}
-
-					bundleTxBz := make([]byte, len(sdkTxBz))
-					copy(bundleTxBz, sdkTxBz)
-					bundledTxBz[index] = sdkTxBz
-				}
+				// if the transaction is already in the (partial) block proposal, we skip it.
+				if proposal.Contains(bundledTxInfo.Hash) {
+					l.Logger().Info(
+						"failed to select auction bid tx for lane; tx is already in proposal",
+						"tx_hash", bundledTxInfo.Hash,
+					)
 
-				// At this point, both the bid transaction itself and all the bundled
-				// transactions are valid. So we select the bid transaction along with
-				// all the bundled transactions. We also mark these transactions as seen and
-				// update the total size selected thus far.
-				txs = append(txs, bidTxBz)
-				txs = append(txs, bundledTxBz...)
+					continue selectBidTxLoop
+				}
 
-				// Write the cache context to the original context when we know we have a
-				// valid bundle.
-				write()
+				// If the bundled transaction is a bid transaction, we skip it.
+				if l.Match(ctx, bundleTx) {
+					l.Logger().Info(
+						"failed to select auction bid tx for lane; bundled tx is another bid transaction",
+						"tx_hash", bundledTxInfo.Hash,
+					)
 
-				break selectBidTxLoop
-			}
+					txsToRemove = append(txsToRemove, bidTx)
+					continue selectBidTxLoop
+				}
 
-			l.Logger().Info(
-				"failed to select auction bid tx for lane; tx size is too large",
-				"tx_size", bidTxSize,
-				"max_size", maxTxBytes,
-			)
-		}
+				if gasLimitSum += bundledTxInfo.GasLimit; gasLimitSum > limit.MaxGasLimit {
```

Comment:
We shld be making the same bytes check for bundledTxs, no?

Reply:
good question but the size of the bid tx already includes the size of all of its bundled transactions so we only need to make this check on the bid tx itself.